scripts/lua2dox.lua

```diff
@@ -201,17 +201,17 @@ local function process_magic(line, generics)
 
   if directive == 'param' then
     for _, type in ipairs(TYPES) do
-      magic = magic:gsub('^param%s+([a-zA-Z_?]+)%s+.*%((' .. type .. ')%)', 'param %1 %2')
-      magic =
-        magic:gsub('^param%s+([a-zA-Z_?]+)%s+.*%((' .. type .. '|nil)%)', 'param %1 %2')
+      magic = magic
+        :gsub('^param%s+([a-zA-Z_?]+)%s+.*%((' .. type .. '|nil)%)', 'param %1 %2') -- verbose optional type
+        :gsub('^param%s+([a-zA-Z_?]+)%s+.*%((' .. type .. '%[%]|nil)%)', 'param %1 %2') -- verbose optional array
+        :gsub('^param%s+([a-zA-Z_?]+)%s+.*%((' .. type .. '[%?]?)%)', 'param %1 %2') -- non-nil or optional type
+        :gsub('^param%s+([a-zA-Z_?]+)%s+.*%((' .. type .. '%[%][%?]?)%)', 'param %1 %2') -- non-nil or optional array
```

Comment:
This is verbose but I couldn't think of a better way to do it. Does anyone have any suggestions?

Reply:
You won't be able to do it using a single pattern.

The best way to handle this is to split the string, process each part separately and then rejoin.